docs/DeveloperGuide.md

```diff
@@ -1,34 +1,561 @@
+TimeTableManager may need user input to be initialised, and so it is placed in a validation loop.
+A welcome message is shown at the end of the initialisation phase.
+
+![Sequence diagram 2 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram2.png?raw=true)
+The main loop is contained in the 'runCommandLoopUntilExitCommand()' method.
+The main loop follows the following steps:
+1. Get the user command as a string.
+2. Parse the command using Parser. Parser will return a Command object.
+3. Execute the Command object with the `execute()` method. This will do the necessary work and return a CommandResult object.
+4. If any data was changed, StateManager will run `saveState()` to facilitate undo commands.
+5. Finally, use the CommandResult object to show the result of the command to the user using TextUi.
+
+Note that the Command and CommandResult objects are destroyed at the end of the method. Additionally, the Parser object
+is destroyed after each use.
+
+![Sequence diagram 3 for Main loop](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainSequenceDiagram3.png?raw=true)
+After exiting the main loop, the main class calls InputOutputManager's saving methods to save all user data.
+After that, the program terminates.
+
+### Add/Delete Feature
+This feature is facilitated by the TaskManager, ModuleManager classes.
+Extending from the abstract Command class are the AddModule, AddTask Command classes. This feature implements the following operations:
+* AddTask - Add a task to the task list through `TaskManager.add()`
+* AddModule - Add a module to the module list through `ModuleManager.add()`
+* DeleteTask - Deletes a task from the task list through `TaskManager.delete()`
+* DeleteModule - Deletes a module from the module list through `ModuleManager.delete()`
+
+![Sequence diagram for AddCommand](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/AddCommandSequenceDiagram.png?raw=true)
+
+As seen from the sequence diagram above, this is the flow of an add command.\
+AddCommand is an abstract class, inheriting from it are AddTaskCommand and AddModuleComand.\
+The AddCommandParser decides to create either AddModuleCommand, AddTaskCommand or IncorrectCommand objects based on the user input.\
+Each of these have an execute() function that creates a CommandResult object that shows the user the result of the command through TextUi, using `showOutputToUser()`
+
+Given below is an example usage scenario and how the add feature behaves at each step.
+
+1. The user launches the application for the first time.
+
+2. The user inputs `add -m CS2101` into ra.VI, as the user wants to note down a module named ‘CS2101’ and add it to their module list.\
+The Ui receives the input as a string. Parser parses the string, and thereafter the AddCommandParser, before creating an AddModuleCommand. 
```

Comment:
> Parser parses the string, and thereafter the AddCommandParser

This phrasing is rather ambiguous, is it implying that the string is parsed twice? What is the result of the first parsing step? Why are there two parsing steps? According to your document it is mentioned that the Parser will create a command, but this section appears to suggest that the Parser just forwards the string to another parser class.

Could this be better explained to reduce ambiguity?